### ISSUE TITLE
Proposal for `no-named-as-default`

### ISSUE THREAD

**Issue by jamestalmage** [import-js/eslint-plugin-import#787]
I've taken to doing this sometimes:

```js
// foo.js
export const foo = 'foo';
export const bar = 'bar';
export default foo;
```

This allows me to import foo in a more versatile way:

```js
import foo from './foo';            // I get the default
import {foo, bar} from './foo';  // In this case - I don't need to remember foo is the default and handle it differently
``` 

Currently `no-named-as-default` would object to the first import above, since I am importing the default export and naming it `foo` which conflicts with a named export. However, in this case, both those exports are the exact same thing, so I don't see the harm.

It seems like we could allow this specific usage through via some static analysis.

**Comment by ljharb** [comment]
Being "more versatile" there is also being less explicit.

Also, you don't ever need to remember that foo is the default; because eslint-plugin-import will tell you.

I would be a strong -1 on allowing for this usage; a value should only be exported from one place in a module.

**Comment by soryy708** [comment]
I'm dismissing this as stale, due to inactivity
